presto-spark-launcher/pom.xml

```diff
@@ -47,7 +47,7 @@
         <dependency>
             <groupId>com.facebook.presto.spark</groupId>
             <artifactId>spark-core</artifactId>
-            <scope>provided</scope>
+            <scope>compile</scope>
```

Comment:
This has to be provided. We are using this dependency only as an API reference compile time. Spark provides this dependency, we don't want to introduce dependency clashes.

Currently we compile with Spark 2.1 and we support up to Spark 2.4. It might not be compatible with Spark 3.1.1, and we didn't look into what is missing to support Spark 3.x.x+